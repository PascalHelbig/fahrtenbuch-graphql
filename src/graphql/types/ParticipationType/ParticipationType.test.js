const { makeExecutableSchema, addMockFunctionsToSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const ParticipationType = require('./ParticipationType');

it('has to be a valid schema', () => {
  const SchemaDefinition = 'schema { query: Participation }';
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, ParticipationType] });
  addMockFunctionsToSchema({
    schema,
    mocks: {
      Participation: () => ({
        id: () => '1337',
        user: () => ({ id: '1338' }),
        boat: () => ({ id: '1339' }),
        entry: () => ({ id: '1340' }),
      }),
    },
  });

  const query = `
    query {
      id
      user { id }
      boat { id }
      entry { id }
    }
  `;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot(),
  );
});
