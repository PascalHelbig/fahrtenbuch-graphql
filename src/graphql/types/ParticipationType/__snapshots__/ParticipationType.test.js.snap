exports[`test has to be a valid schema 1`] = `
Object {
  "data": Object {
    "boat": Object {
      "id": "1339",
    },
    "entry": Object {
      "id": "1340",
    },
    "id": "1337",
    "user": Object {
      "id": "1338",
    },
  },
}
`;
